core/src/test/java/org/infinispan/util/BaseControlledConsistentHashFactory.java

```diff
@@ -19,27 +22,38 @@
  * @since 6.0
  */
 @SuppressWarnings("unchecked")
-public abstract class BaseControlledConsistentHashFactory implements ConsistentHashFactory<DefaultConsistentHash>,
+public abstract class BaseControlledConsistentHashFactory<CH extends ConsistentHash> implements ConsistentHashFactory<CH>,
```

Comment:
That would require ScatteredConsistentHash to extend DefaultConsistentHash - are you suggesting that? If one extended the other, there would be fields left unused, and methods completely overridden.